### PR TITLE
Fixes #22188: Changes made by validated users still need to be validated

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/22188

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
PR updated with a new commit

File: change-validation/src/main/scala/bootstrap/rudder/plugin/ChangeValidationConf.scala [hunk 1]
```diff
@@ -134,13 +135,32 @@ class ChangeValidationWorkflowLevelService(
    * Methode to use to combine several validationNeeded check. It's the same for all objects?
```
Comment (ncharles) [inline change-validation/src/main/scala/bootstrap/rudder/plugin/ChangeValidationConf.scala:135]: you should probably add a comment there saying that it always skips workflow for validated user

**Comment by ElaadF** [comment]
PR updated with a new commit

File: change-validation/src/main/scala/bootstrap/rudder/plugin/ChangeValidationConf.scala [hunk 2]
```diff
@@ -131,16 +132,36 @@ class ChangeValidationWorkflowLevelService(
...
+    /*
+     * Here we check if there is a validated user that should not be subject to any validation workflow
+     * if there is no validated user, we iterate over `checks: Seq[ValidationNeeded]` to verify if there
+     * there is a specific workflow.
+     *
+     * Check why we decided to separate the validated user logic from `ValidationNeeded` objects :
+     * https://issues.rudder.io/issues/22188#note-5
+     */
+    validatedUserRepo.get(actor) match {
+      case Full(ea) =>
```
Comment (VinceMacBuche) [inline change-validation/src/main/scala/bootstrap/rudder/plugin/ChangeValidationConf.scala:158]: You should at least log the error in the emptybox case and i do even think it should be an error

Also you can do both match in one go

validatedUserRepo.get(actor) match {
  case Full(Some(_)
  case Full(None)
  case eb : Emptybox => eb  (i think it should be an error (or ar least log here then getWorkflowAux, but I think it's an error case)

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
The bug here was that UserValidationNeeded was not  used.

**Comment by ElaadF** [comment]
OK, squash merging this PR

**Pull request merged**